prisma-fmt/src/hover.rs

```diff
@@ -0,0 +1,245 @@
+            let enm = ctx.db.walk((ctx.initiating_file_id, enum_id)).ast_enum();
+            Some(hover_enum(enm, name))
+        }
+        SchemaPosition::CompositeType(ct_id, CompositeTypePosition::Name(name)) => {
+            let ct = ctx.db.walk((ctx.initiating_file_id, ct_id)).ast_composite_type();
+            Some(hover_composite(ct, name))
+        }
+
+        // --- Block Field Names ---
+        SchemaPosition::Model(model_id, ModelPosition::Field(field_id, FieldPosition::Name(name))) => {
+            let field = ctx
+                .db
+                .walk((ctx.initiating_file_id, model_id))
+                .field(field_id)
+                .ast_field();
+
+            Some(format_hover_content(
+                field.documentation().unwrap_or_default(),
+                "field",
+                name,
+                None,
+            ))
+        }
+        SchemaPosition::CompositeType(ct_id, CompositeTypePosition::Field(field_id, FieldPosition::Name(name))) => {
+            let field = ctx.db.walk((ctx.initiating_file_id, ct_id)).field(field_id).ast_field();
+
+            Some(format_hover_content(
+                field.documentation().unwrap_or_default(),
+                "field",
+                name,
+                None,
+            ))
+        }
+        SchemaPosition::Enum(enm_id, EnumPosition::Value(value_id, EnumValuePosition::Name(name))) => {
+            let value = ctx
+                .db
+                .walk((ctx.initiating_file_id, enm_id))
+                .value(value_id)
+                .ast_value();
+
+            Some(format_hover_content(
+                value.documentation().unwrap_or_default(),
+                "value",
+                name,
+                None,
+            ))
+        }
+
+        // --- Block Field Types ---
+        SchemaPosition::Model(model_id, ModelPosition::Field(field_id, FieldPosition::Type(name))) => {
+            let initiating_field = &ctx.db.walk((ctx.initiating_file_id, model_id)).field(field_id);
+
+            initiating_field.refine().and_then(|field| match field {
+                walkers::RefinedFieldWalker::Scalar(scalar) => match scalar.scalar_field_type() {
+                    ScalarFieldType::CompositeType(_) => {
+                        let ct = scalar.field_type_as_composite_type().unwrap().ast_composite_type();
+                        Some(hover_composite(ct, ct.name()))
+                    }
+                    ScalarFieldType::Enum(_) => {
+                        let enm = scalar.field_type_as_enum().unwrap().ast_enum();
+                        Some(hover_enum(enm, enm.name()))
+                    }
+                    _ => None,
+                },
+                walkers::RefinedFieldWalker::Relation(rf) => {
+                    let opposite_model = rf.related_model();
+                    let relation_info = rf.opposite_relation_field().map(|rf| (rf, rf.ast_field()));
```

Comment:
Can you explain when the `opposite_relation_field` may be `None`? I'm trying to understand when would `format_relation_info` return empty strings

Reply:
Lets say we have the following schema
```prisma
model interm {
    id Int @id

    forumId Int
    forum   Forum @relation(fields: [forumId], references: [id])
}


/// Forum Doc
model Forum {
    id Int @id
}
```

The opposite relation is yet to be defined in the model `Forum` but we still want to be able to hover over where it's used as a field type in `interm` and see it's model documentation